### PR TITLE
EHPR-25 | Setup test env and deployment pipeline

### PR DESCRIPTION
**Pull request by fwpushan**
- Test Env infra code
- Cleanup 
- Pipeline

### CODE REVIEW

File: terraform/api.tf [hunk 1]
```diff
@@ -75,16 +74,15 @@ resource "aws_apigatewayv2_stage" "api" {
   name        = "$default"
   auto_deploy = true
 
-  access_log_settings {
+  /*access_log_settings {
```
Comment (arranfw) [inline terraform/api.tf:77]: Let's just delete this if we aren't using it, we can always refer to history if we need it again

File: .github/workflows/deploy-test.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
+name: Deployment to aws test environment
```
Comment (arranfw) [inline .github/workflows/deploy-test.yml:1]: Wondering if there's any way to dedupe this deployment code, can we extract the tag name somehow? Tracking changes between our deployment workflows will get tough as time goes on

**Review by arranfw — CHANGES_REQUESTED** [review]
Happy to push the workflow conversation into the new year

I think we should remove unneeded code rather than comment it out

File: apps/api/src/app.controller.spec.ts [hunk 3]
```diff
@@ -16,7 +16,7 @@ describe('AppController', () => {
 
   describe('root', () => {
     it('should return "Hello World!"', () => {
```
Comment (arranfw) [inline apps/api/src/app.controller.spec.ts:18]: can we update the test name too

File: .github/workflows/deploy-test.yml (hunk 2 above)
Comment (fwpushan) [reply .github/workflows/deploy-test.yml:1]: we have different environment setup for this pipelines...the branch deployment will cause issue ...

Comment (fwpushan) [reply .github/workflows/deploy-test.yml:1]: we can not have conditional code in `environment.name`

File: terraform/api.tf (hunk 1 above)
Comment (fwpushan) [reply terraform/api.tf]: Fixed

**Review by arranfw — APPROVED** [review]

**Review by dbayly-freshworks — APPROVED** [review]
I don't see any red flags.  👍

**Pull request merged**